sw/ground_segment/python/sdlog_download_app/sdlogdownloadframe.py

```diff
@@ -0,0 +1,220 @@
+        self.stopButton.Disable()
+        self.downloadButton = wx.Button(self, id=3, label="Download!", pos=wx.Point(190, 150), size=wx.Size(100, 25))
+        self.Bind(wx.EVT_BUTTON, self.onButton)
+
+        sizer = wx.BoxSizer(wx.HORIZONTAL)
+        self.SetSizer(sizer)
+        sizer.Layout()
+
+        # Some variables
+        self.download_counter = 0
+        self.download_available = 0
+        self.last_command = 0
+        self.unique_id = 0
+        self.download_timer = threading.Timer
+        self.request_timer = threading.Timer
+        self.timeout_time = 0.1
+
+    # Called on button push
+    def onButton(self, event):
+        button = event.GetEventObject()
+        index = int(button.GetId())
+        self.last_command = index
+        setting_index = self.settings.name_lookup["sdlogger.cmd"].index
+        unique_id_idx = self.settings.name_lookup["sdlogger.unique_id"].index
+        if index == 1:
+            # First set unique id
+            IvySendMsg("dl DL_SETTING %s %s %s" % (self.ac_id, unique_id_idx, int(time.time()) - 1428500000))
+            self.last_command = 0
+            # Continuing trough callback from setting unique id
+        elif index == 2:
+            IvySendMsg("dl DL_SETTING %s %s %s" % (self.ac_id, setting_index, 2))
+            self.stopButton.Disable()
+            self.request_timer = threading.Timer(0.5, self.stopButton.Enable)
+            self.request_timer.start()
+        elif index == 3:
+            IvySendMsg("dl DL_SETTING %s %s %s" % (self.ac_id, setting_index, 3))
+            self.downloadButton.Disable()
+
+    def OnStatusRequest(self, event):
+        setting_index = self.settings.name_lookup["sdlogger.cmd"].index
+        IvySendMsg("dl DL_SETTING %s %s %s" % (self.ac_id, setting_index, 3))
+
+    def OnLogPacketReceive(self, agent, *larg):
+        wx.CallAfter(self.process_log_packet, larg[0])
+
+        # Called to update GUI with new values
+    def process_log_packet(self, message):
+        self.inDataLabel.SetLabel(message)
+        # Extract field values
+        message_values = list(filter(None, message.split(' ')))
+        message_values = message_values[0:1] + message_values[2:]
+        if self.last_command == 3:
+            self.download_counter = 0
+            self.download_available = int(message_values[1])
+            self.unique_id = message_values[3]
+            self.statusDataLabel.SetLabel(message)
+            # Set the unique ID before reading, otherwise zeros will be returned
+            setting_index = self.settings.name_lookup["sdlogger.unique_id"].index
+            IvySendMsg("dl DL_SETTING %s %s %s" % (self.ac_id, setting_index, self.unique_id))
+            #self.RequestNextPacket() do this in callback instead
+        elif self.last_command == 57:
+            self.download_timer.cancel()
+            self.inDataLabel.SetLabel(message)
+            fh = open("logfile.txt", "a")
```

Comment:
Would be better to have the log file name as a parameter, or even better, chosen from a selection box.